Build warning for 64 bit platforms in lwm2m tests

**Describe the bug**
Commit b1d24e425c3a30 introduced a build warning in `lwm2m_obj_device.c`, related to a `LOG_ERR` statement with an incorrect format specifier. The previously used format specifier was `%u` for a variable of type `size_t`, which is correct for 32 bit systems but not portable to 64 bit ones. This results in a build warning on 64 bit platforms.

**To Reproduce**
Steps to reproduce the behavior:
1. west build -p always -b intel_socfpga_agilex5_socdk tests/net/lib/lwm2m/block_transfer/
2. See build warning

**Expected behavior**
Build should pass without warnings.

**Impact**
Blocking CI on #64036

**Logs and console output**
Here is the build warning:
```
/home/danieldegrasse/zephyr/zephyrproject/zephyr/subsys/net/lib/lwm2m/lwm2m_obj_device.c:264:33: warning: format '%u' expects argument of type 'unsigned int', but argument 2 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]
  264 |                         LOG_ERR("Error code list too large: %u", len);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~
      |                                                                  |
      |                                                                  size_t {aka long unsigned int}
/home/danieldegrasse/zephyr/zephyrproject/zephyr/include/zephyr/logging/log_core.h:275:42: note: in definition of macro 'Z_LOG2'
  275 |                 z_log_printf_arg_checker(__VA_ARGS__); \
      |                                          ^~~~~~~~~~~
/home/danieldegrasse/zephyr/zephyrproject/zephyr/include/zephyr/logging/log.h:42:25: note: in expansion of macro 'Z_LOG'
   42 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
/home/danieldegrasse/zephyr/zephyrproject/zephyr/subsys/net/lib/lwm2m/lwm2m_obj_device.c:264:25: note: in expansion of macro 'LOG_ERR'
  264 |                         LOG_ERR("Error code list too large: %u", len);
      |                         ^~~~~~~
/home/danieldegrasse/zephyr/zephyrproject/zephyr/subsys/net/lib/lwm2m/lwm2m_obj_device.c:264:62: note: format string is defined here
  264 |                         LOG_ERR("Error code list too large: %u", len);
      |                                                             ~^
      |                                                              |
      |                                                              unsigned int
      |                                                             %lu
```

--- a/subsys/net/lib/lwm2m/lwm2m_obj_device.c
+++ b/subsys/net/lib/lwm2m/lwm2m_obj_device.c
@@ -261,7 +261,7 @@ static int lwm2m_obj_device_settings_set(const char *name, size_t len,
 
 	if (settings_name_steq(name, ERROR_LIST_KEY, &next) && !next) {
 		if (len > sizeof(error_code_list)) {
-			LOG_ERR("Error code list too large: %u", len);
+			LOG_ERR("Error code list too large: %zu", len);
 			return -EINVAL;
 		}
 

